### ISSUE TITLE
Setting Context Path does not work since version 3.5 anymore

### ISSUE THREAD

**Issue by patbaumgartner** [DependencyTrack/dependency-track#401]
### Current Behavior:
Running the Docker container and passing after java -jar dt-embedded.war -context /dtrack as describted in the documentation does not work anymore after an upgrade from 3.4.1

### Steps to Reproduce:

java -Xmx4G -jar dependency-track-embedded.war -context /dtrack

### Expected Behavior:

Serving the Login page after calling http://localhost:8080/dtrack

### Environment:
  
  - Dependency-Track Version: 3.5.1
  - Distribution: Docker | Embedded WAR
  - BOM Format & Version: Cyclone
  - Database Server:  H2 | PostgreSQL 
  - Browser:  Chrome and Firefox

### Additional Details:

Seems to be an Alpine or jetty issue. But don't know when this was introduced.

**Comment by stevespringett** [comment]
The Docker container docs does not mention changing the context path.
https://docs.dependencytrack.org/getting-started/deploy-docker/

The executable war docs do mention it
https://docs.dependencytrack.org/getting-started/deploy-exewar/

Modifying the context was never supported in the Docker container.

**Comment by patbaumgartner** [comment]
Hi @stevespringett 

I am aware that context-path is not supported in docker. I exended your container with a different CMD. I am sure it's not a docker problem since I can reproduce it also just from the command line too.

**Steps to reproduce:** 

I removed "rm -rf ~/.dependency-track"

I downloaded  the war from - https://github.com/DependencyTrack/dependency-track/releases/download/3.5.1/dependency-track-embedded.war

I copied "java -Xmx4G -jar dependency-track-embedded.war -context /dtrack" from the documentation - https://docs.dependencytrack.org/getting-started/deploy-exewar/#examples

and ran it on my bash.

I opened my browser (and recorded the requests) on http://localhost:8080/dtrack and saw a redirect to http://localhost:8080/dtrack/ which was answered with a 404 ... (saw the same behavior on a browser with a clean profile to make sure that the redirect was not cached)

Running it without the "-context /dtrack" works and the page gets delivered on localhost:8080

**Comment by stevespringett** [comment]
Interesting... So there was some changes in the WhitelistUrlFilter configuration defined in web.xml in v3.5.0. The changes were necessary to prevent access to server resources that should not be accessible (because it's an executable war). The changes seem to have also caused contexts other than the root (/) to not work.

```
    <filter>
        <filter-name>WhitelistUrlFilter</filter-name>
        <filter-class>alpine.filters.WhitelistUrlFilter</filter-class>
        <init-param>
            <param-name>allowUrls</param-name>
            <param-value>/index.jsp,/functions.js,/admin,/api,/assets,/component,/components,/favicon.ico,/images,/license,/licenses,/project,/projects,/style.css,/vulnerabilities,/vulnerability</param-value>
        </init-param>
    </filter>
    <filter-mapping>
        <filter-name>WhitelistUrlFilter</filter-name>
        <url-pattern>/*</url-pattern>
    </filter-mapping>
```

As a workaround, you could change the configuration to:

```
    <filter>
        <filter-name>WhitelistUrlFilter</filter-name>
        <filter-class>alpine.filters.WhitelistUrlFilter</filter-class>
        <init-param>
            <param-name>allowUrls</param-name>
            <param-value>/dtrack/index.jsp,/dtrack/functions.js,/dtrack/admin,/dtrack/api,/dtrack/assets,/dtrack/component,/dtrack/components,/dtrack/favicon.ico,/dtrack/images,/dtrack/license,/dtrack/licenses,/dtrack/project,/dtrack/projects,/dtrack/style.css,/dtrack/vulnerabilities,/dtrack/vulnerability</param-value>
        </init-param>
    </filter>
    <filter-mapping>
        <filter-name>WhitelistUrlFilter</filter-name>
        <url-pattern>/*</url-pattern>
    </filter-mapping>
```

**Comment by stevespringett** [comment]
This has been fixed in Alpine 1.5.1 and will be included in Dependency-Track v3.6.0

**Comment by patbaumgartner** [comment]
Thank you very much!
